### PR TITLE
Add git init script when initializing project

### LINKED ISSUE #670: Init: call git init
When `init` the project, it would be nice to call git init, at the end of this workflow

suggested by @fdelbrayelle

### PR DESCRIPTION
**Pull request by matthieulapatate**
fix #670 
Directory is generated but return an error while working.

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
It's because you didn't respect the architecture. See the errors in logs:

```
Error:  tech.jhipster.lite.HexagonalArchTest.domainShouldNotInteractWithOutside  Time elapsed: 0.786 s  <<< FAILURE!
java.lang.AssertionError: 
Architecture Violation [Priority: MEDIUM] - Rule 'no classes that reside in any package ['..domain..'] should depend on classes that reside in any package ['..application..', '..infrastructure..', '..primary..', '..secondary..'], because Domain should not interact with application, infrastructure, primary or secondary' was violated (1 times):
Constructor <tech.jhipster.lite.generator.init.domain.InitDomainService.<init>(tech.jhipster.lite.generator.packagemanager.npm.domain.NpmService, tech.jhipster.lite.generator.project.domain.ProjectRepository, java.util.concurrent.Executor)> references class object <tech.jhipster.lite.generator.packagemanager.npm.infrastructure.secondary.NpmLocalRepository> in (InitDomainService.java:35)
	at tech.jhipster.lite.HexagonalArchTest.domainShouldNotInteractWithOutside(HexagonalArchTest.java:188)

Error:  tech.jhipster.lite.HexagonalArchTest.shouldNotHaveFrameworkDependenciesInContext  Time elapsed: 0.04 s  <<< FAILURE!
java.lang.AssertionError: 
Architecture Violation [Priority: MEDIUM] - Rule 'classes that reside in any package ['tech.jhipster.lite.generator.init.domain..'] should only depend on classes that reside in any package ['tech.jhipster.lite.generator.init.domain..', 'java..', 'org.slf4j..', 'org.apache.commons.lang3..', 'tech.jhipster.lite.error..', 'tech.jhipster.lite.generator.server.springboot.cache.common..', 'tech.jhipster.lite.generator.server.springboot.database.sqlcommon..', 'tech.jhipster.lite.generator.server.springboot.springcloud.common..', 'tech.jhipster.lite.generator.server.springboot.common..', 'tech.jhipster.lite.generator.packagemanager.npm..', 'tech.jhipster.lite.generator.project..', 'tech.jhipster.lite.generator.buildtool.generic..', 'tech.jhipster.lite.generator.buildtool.gradle..', 'tech.jhipster.lite.generator.buildtool.maven..', 'tech.jhipster.lite.common..'], because Domain model should only depend on himself and a very limited set of external dependencies' was violated (1 times):
Field <tech.jhipster.lite.generator.init.domain.InitDomainService.timeout> is annotated with <org.springframework.beans.factory.annotation.Value> in (InitDomainService.java:0)
	at tech.jhipster.lite.HexagonalArchTest.lambda$shouldNotHaveFrameworkDependenciesInContext$12(HexagonalArchTest.java:121)
	at tech.jhipster.lite.HexagonalArchTest.shouldNotHaveFrameworkDependenciesInContext(HexagonalArchTest.java:113)
```

Then, you don't need to add all these codes as the `ProjectRepository` contains already the method `gitInit`

File: a [hunk 1]
```diff
@@ -0,0 +1,14 @@
+[INFO] Scanning for projects...
```
Comment (pascalgrimaud) [inline a:1]: why do you commit this file ?

File: src/main/java/tech/jhipster/lite/generator/init/domain/InitDomainService.java [hunk 2]
```diff
@@ -33,6 +33,7 @@ public void init(Project project) {
     addEditorConfiguration(project);
     addPrettier(project);
     addPackageJson(project);
+    addGitInit(project);
```
Comment (pascalgrimaud) [inline src/main/java/tech/jhipster/lite/generator/init/domain/InitDomainService.java:36]: it is not tested by unit test or integration test

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

**Comment by pascalgrimaud** [comment]
@matthieulapatate : still waiting your squash, to have 1 single commit

**Comment by matthieulapatate** [comment]
> @matthieulapatate : still waiting your squash, to have 1 single commit

Might be good right ?

**Comment by pascalgrimaud** [comment]
good, it can be merged, thanks @matthieulapatate

**Review by pascalgrimaud — APPROVED** [review]

**Pull request merged**